app/src/main/java/com/github/k409/fitflow/ui/screens/hydration/HydrationScreen.kt

```diff
@@ -0,0 +1,156 @@
+import com.github.k409.fitflow.ui.common.Dialog
+import com.github.k409.fitflow.ui.common.NumberPicker
+import com.github.k409.fitflow.ui.common.PickerState.Companion.rememberPickerState
+import com.github.k409.fitflow.ui.components.WaterIndicator
+import com.github.k409.fitflow.ui.components.hydration.WaterIntakeLog
+
+@Composable
+fun WaterLoggingScreen(
+    viewModel: HydrationViewModel = hiltViewModel(),
+) {
+    val context = LocalContext.current
+    val uiState by viewModel.uiState.collectAsState()
+
+    val editCupSizeDialogState = remember { mutableStateOf(false) }
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(horizontal = 16.dp),
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .fillMaxHeight(0.6f),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center,
+        ) {
+            WaterIndicator(
+                totalWaterAmount = uiState.dailyGoal,
+                usedWaterAmount = uiState.today,
+            )
+
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 32.dp),
+                contentAlignment = Alignment.Center,
+            ) {
+                FilledTonalButton(
+                    modifier = Modifier.align(Alignment.Center),
+                    border = ButtonDefaults.outlinedButtonBorder,
+                    onClick = {
+                        viewModel.addWaterCup()
+                        HydrationReminder().scheduleWaterReminder(context)
+                    },
+                ) {
+                    Text(stringResource(R.string.drink_ml, uiState.cupSize))
+                }
+
+                IconButton(
+                    modifier = Modifier.align(Alignment.CenterEnd),
+                    onClick = {
+                        editCupSizeDialogState.value = true
+                    },
+                ) {
+                    Icon(
+                        modifier = Modifier
+                            .size(18.dp)
+                            .alpha(0.5f),
+                        imageVector = Icons.Outlined.ModeEditOutline,
+                        contentDescription = null,
+                    )
+                }
+            }
+        }
+
+        WaterIntakeLog(
+            milliliters = uiState.yesterday.toLong(),
+            thisWeek = uiState.thisWeek.toDouble(),
+            thisMonth = uiState.thisMonth.toDouble(),
+        )
+
+        if (editCupSizeDialogState.value) {
+            EditCupSizeDialog(
+                onDismiss = { editCupSizeDialogState.value = false },
+                onSaveClick = { viewModel.setCupSize(it) },
+                currentCupSize = uiState.cupSize,
+            )
+        }
+    }
```

Comment:
There already exists a numberPickerDialog function inside Dialogs.kt file. I am pretty sure that you can reuse that instead of your custom number picker class and remove a lot of unnecessary code this way. Here is my suggestion:
```suggestion
        if (editCupSizeDialogState.value) {
            val cupValues = Array(300) { (it * 10 + 10).toString() }
            NumberPickerDialog(
                onDismissRequest = { editCupSizeDialogState.value = false },
                onConfirmation = { editCupSizeDialogState.value = false
                    viewModel.setCupSize(it.toInt()*10) },
                dialogTitle = stringResource(R.string.cup_size_ml),
                dialogText = "",
                minValue = 1,
                maxValue = 300,
                initialValue = uiState.cupSize/10,
                displayedValues = cupValues
            )
        }
```